lib/modules/manager/circleci/schema.ts

```diff
@@ -9,9 +9,33 @@ export const CircleCiJob = z.object({
   docker: z.array(CircleCiDocker).optional(),
 });
 
+type Orb = {
+  executors?: Record<string, CircleCiJob>;
+  jobs?: Record<string, CircleCiJob>;
+  orbs?: Record<string, string | Orb>;
+};
+
+type CircleCiJobInput = z.input<typeof CircleCiJob>;
+
+type OrbInput = {
+  executors?: Record<string, CircleCiJobInput>;
+  jobs?: Record<string, CircleCiJobInput>;
+  orbs?: Record<string, string | OrbInput>;
+};
+
+export type CircleCiOrb = z.infer<typeof CircleCiOrb>;
+export const CircleCiOrb: z.ZodType<Orb, any, OrbInput> = z.object({
```

Comment:
```suggestion
export const CircleCiOrb = z.object({
```

Why this extra type cast?

Reply:
without it, the type inference for the Orb input/output seems to fail
```
$ git diff
diff --git a/lib/modules/manager/circleci/schema.ts b/lib/modules/manager/circleci/schema.ts
index 660fe14d8..45c7556e2 100644
--- a/lib/modules/manager/circleci/schema.ts
+++ b/lib/modules/manager/circleci/schema.ts
@@ -21,7 +21,7 @@ type OrbInput = {
   orbs?: Record<string, string | OrbInput>;
 };

-export const CircleCiOrb: z.ZodType<Orb, any, OrbInput> = z.object({
+export const CircleCiOrb = z.object({
   executors: z.record(z.string(), CircleCiJob).optional(),
   jobs: z.record(z.string(), CircleCiJob).optional(),
   orbs: z.lazy(() =>

$ pnpm build
[...]

lib/modules/manager/circleci/extract.ts:34:9 - error TS2322: Type 'unknown[]' is not assignable to type '{ docker?: { image: string; }[] | undefined; }[]'.
  Type 'unknown' is not assignable to type '{ docker?: { image: string; }[] | undefined; }'.

34   const environments: CircleCiJob[] = [
           ~~~~~~~~~~~~

lib/modules/manager/circleci/schema.ts:12:6 - error TS6196: 'Orb' is declared but never used.

12 type Orb = {
        ~~~

lib/modules/manager/circleci/schema.ts:18:6 - error TS6196: 'OrbInput' is declared but never used.

18 type OrbInput = {
        ~~~~~~~~

lib/modules/manager/circleci/schema.ts:24:14 - error TS7022: 'CircleCiOrb' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.

24 export const CircleCiOrb = z.object({
                ~~~~~~~~~~~

lib/modules/manager/circleci/schema.ts:27:16 - error TS7024: Function implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.

27   orbs: z.lazy(() =>
                  ~~~~~
28     z.record(z.string(), z.union([z.string(), CircleCiOrb])).optional(),
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

(https://zod.dev/?id=zodtype-with-zodeffects?)